packages/template-retail-react-app/app/pages/cart/index.jsx

```diff
@@ -141,31 +188,43 @@ const Cart = () => {
             setSelectedItem(undefined)
         }
     }
+    /***************************** Update Cart **************************/
 
+    /***************************** Update quantity **************************/
     const changeItemQuantity = debounce(async (quantity, product) => {
         // This local state allows the dropdown to show the desired quantity
         // while the API call to update it is happening.
+        const previousQuantity = localQuantity[product.itemId]
         setLocalQuantity({...localQuantity, [product.itemId]: quantity})
         setCartItemLoading(true)
         setSelectedItem(product)
-        try {
-            const item = {
-                productId: product.id,
-                quantity: parseInt(quantity)
+
+        await updateItemInBasketAction.mutate(
+            {
+                parameters: {basketId: basket?.basketId, itemId: product.itemId},
+                body: {
+                    productId: product.id,
+                    quantity: parseInt(quantity)
+                }
+            },
+            {
+                onSuccess: () => {
+                    // reset the state
+                    setCartItemLoading(false)
+                    setSelectedItem(undefined)
+                    setLocalQuantity({...localQuantity, [product.itemId]: undefined})
+                },
+                onError: () => {
+                    // reset the quantity to the previous value
+                    setLocalQuantity({...localQuantity, [product.itemId]: previousQuantity})
+                    showError()
+                }
             }
-            await basket.updateItemInBasket(item, product.itemId)
-        } catch {
-            showError()
-        } finally {
-            // reset the state
-            setCartItemLoading(false)
-            setSelectedItem(undefined)
-            setLocalQuantity({...localQuantity, [product.itemId]: undefined})
-        }
```

Comment:
Same problem as I pointed out in a different spot, dropping `try / catch / finally` means you need a functional replacement for `finally`

Reply:
I added onSettled to reset the state. Please have another review if you are available. Thanks